torch/futures/__init__.py

```diff
@@ -81,6 +81,40 @@ def then(self, callback):  # type: (Callable[[Future[T]], S]) -> Future[S]
         """
         return cast(Future[S], super().then(callback))
 
+    # Have to use string annotations because  PEP-0563 is not available in 3.6
+    def add_done_callback(self, callback):  # type: (Callable[[Future[T]], None]) -> None
```

Comment:
As @pritamdamania87 suggested, let's make this API private by adding a "_" prefix to it.

Reply:
done